### PR TITLE
ENTESB-4090 - Fixing fuse integration (Fuse hack breaks kie-remote-client usage in Fuse code)

### PR DESCRIPTION
**Pull request by mrietveld**
Related to https://github.com/droolsjbpm/drools/pull/516

See the [drools PR](https://github.com/droolsjbpm/drools/pull/516) for the full explanation. 

(as well as PR's on droolsjbpm-integration and fuse-bxms-integ).

### CODE REVIEW

File: kie-api/src/main/java/org/kie/api/runtime/rule/QueryResultsRow.java [hunk 1]
```diff
@@ -21,4 +21,18 @@
...
+     */
+    public Object get(String identifier);
+
+    /**
+     * Return the {@link FactHandle} for the given Declaration identifier.
+     * @param identifier
+     * @return the {@link FactHandle} instance
+     */
+    public FactHandle getFactHandle(String identifier);
+
```
Comment (psiroky) [inline kie-api/src/main/java/org/kie/api/runtime/rule/QueryResultsRow.java]: Question about backwards compatibility: I am assuming the interface `QueryResultsRow` should _not_ be directly implemented by end-users, they will only use concrete instances of classes implementing this interface. So adding new methods to the interface should not actually break backwards compatibility. Is that correct?

Comment (mrietveld) [inline kie-api/src/main/java/org/kie/api/runtime/rule/QueryResultsRow.java]: Correct: 1. end-users should _not_ be implementing this interface and 2. adding new methods willnot (AFAIK) break backwards compatibility. The underlying implementations already were implementing both of these methods.

Comment (psiroky) [inline kie-api/src/main/java/org/kie/api/runtime/rule/QueryResultsRow.java]: thanks @mrietveld, just wanted to double check.

**Comment by mariofusco** [comment]
-1
Those methods don't seem strictly necessary in public api for now.

**Pull request closed without merging**